### PR TITLE
Add time series MIA examples

### PR DESCRIPTION
**Pull request by fazelehh**
## Summary
- Add time series MIA examples from Nicolas/Tobias thesis work
- Support 7 forecasting models: LSTM, TCN, DLinear, NBeats, WaveNet, NHiTS, Bhowmick
- Support 4 datasets: ECG, EEG, ELD, LCL
- Include IndividualizedInputHandler for individual-level time series data

## Changes
- New example directory: `examples/mia/time_series_mia/`
- Data preprocessing with multiple scaling options
- Evaluation metrics (MSE, MAE, SMAPE, ND, etc.)
- Early stopping support

## Fixes applied during review
- Fixed bug in `model_preparation.py` (undefined `val_loss` variable)
- Fixed deprecated `np.float` in `NBeats.py` → `np.float64`
- Fixed typo: renamed `Bromick` → `Bhowmick`
- Added missing `__init__.py` files
- Fixed `set_seed.py` for true reproducibility

### CODE REVIEW

File: examples/mia/time_series_mia/utils/metrics.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import numpy as np
+
+def nd(target, pred): # Normalized deviation
+    numerator = np.sum(np.abs(target - pred))
+    denominator = np.sum(np.abs(target))
+    return numerator / denominator if denominator != 0 else np.nan
+
+def mse(target, pred):
```
Comment (henrikfo) [inline examples/mia/time_series_mia/utils/metrics.py]: Do we need mse, rmse etc. when they are simple calls from the numpy module?

File: examples/mia/time_series_mia/utils/model_preparation.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    if optimizer_name.lower() == "adam":
+        optimizer = optim.Adam(model.parameters())
+    elif optimizer_name.lower() == "sgd":
+        optimizer = optim.SGD(model.parameters())
+    else:
+        raise NotImplementedError(f"Optimizer not found: {optimizer_name}")
+    
+
+    best_val_loss = (-1, np.inf)  # (epoch, validation loss)
+    best_state_dict = model.state_dict()
```
Comment (henrikfo) [inline examples/mia/time_series_mia/utils/model_preparation.py]: For all `best_state_dict = model.state_dict()` use `best_state_dict = copy.deepcopy(model.state_dict())` instead. Otherwise there's a risk that the wrong state_dict() will get saved.

File: examples/mia/time_series_mia/main.ipynb [hunk 3]
```diff
@@ -0,0 +1,324 @@
+{
```
[the remark this answers is not in the document]
Comment (fazelehh) [reply examples/mia/time_series_mia/main.ipynb]: Updated leakpro/utils/seed.py (benchmark=False), deleted local set_seed.py, replaced all set_seed(...) calls with seed_everything(...) in main.py and main.ipynb

File: examples/mia/time_series_mia/utils/model_preparation.py (hunk 2 above)
Comment (fazelehh) [reply examples/mia/time_series_mia/utils/model_preparation.py]: Added import copy and applied copy.deepcopy(model.state_dict()) in both model_preparation.py (2 places) and data_handler.py (2 places)
